src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2584,6 +2584,18 @@ internal bool IsMoveableVariable(BoundExpression expr, out Symbol accessedLocalO
                             expr = receiver;
                             continue;
                         }
+                    case BoundKind.InlineArrayAccess:
+                        {
+                            var elementAccess = (BoundInlineArrayAccess)expr;
+
+                            if (elementAccess.GetItemOrSliceHelper is WellKnownMember.System_Span_T__get_Item or WellKnownMember.System_ReadOnlySpan_T__get_Item)
+                            {
+                                expr = elementAccess.Expression;
+                                continue;
```

Comment:
Should the behavior be detailed in the speclet?

Reply:
> Should the behavior be detailed in the speclet?

https://github.com/dotnet/csharplang/pull/7362